dds/DCPS/MultiTopicDataReaderBase.cpp

```diff
@@ -278,8 +326,12 @@ void MultiTopicDataReaderBase::cleanup()
   }
   DataReaderImpl* dri = dynamic_cast<DataReaderImpl*>(resulting_reader_.in());
   SubscriberImpl* si = dynamic_cast<SubscriberImpl*>(sub.in());
-  si->remove_from_datareader_set(dri);
-  dri->cleanup();
+  if (si && dri) {
```

Comment:
Yea.  Should've done it that way.  I believe I was contemplating what to do if !si or !dri happened so I set it up that way initially.  I'll nest the ifs.